flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -291,7 +291,10 @@ def add_floodwall(self, floodwall: FloodWallModel, measure_path=Path):
 
         # Add floodwall attributes to geodataframe
         gdf_floodwall["name"] = floodwall.name
-        gdf_floodwall["z"] = floodwall.elevation.convert(UnitTypesLength("meters"))
+        if (gdf_floodwall.geometry.type == "MultiLineString").any():
+            gdf_floodwall = gdf_floodwall.explode()
+        if "z" not in gdf_floodwall.columns:
+            gdf_floodwall["z"] = floodwall.elevation.convert(UnitTypesLength("meters"))
```

Comment:
This would mean that if "z" is already in the columns you will not convert the units.

Reply:
Good one, changed this.